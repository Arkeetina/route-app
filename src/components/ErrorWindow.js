import React from 'react';

const ErrorWindow = props => (
  <div className="box-layout__box">
    <p>{props.errorMessage}</p>
  </div>
);

export default ErrorWindow;