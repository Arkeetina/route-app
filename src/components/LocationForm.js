import React, {Component} from 'react';

export default class LocationForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      startOffLocation: '',
      dropOffLocation: '',
      error: '',
      disabled: true,
    }
  }

  onStartOffLocationChange = event => {
    const startOffLocation = event.target.value;
    this.setState(() => ({ startOffLocation }));
  }

  onDropOffLocationChange = event => {
    const dropOffLocation = event.target.value;
    this.setState(() => ({ dropOffLocation }));
  }

  onSubmit = event => {
    event.preventDefault();
    if (!this.state.startOffLocation.trim() || !this.state.dropOffLocation.trim()) {
      this.setState(() => ({ error: 'Please provide start and drop off locations' }));
    } else {
      this.setState(() => ({ error: '' }));
      this.setState(() => ({ disabled: true }));
      let startOffLocation = this.state.startOffLocation;
      let dropOffLocation = this.state.dropOffLocation;
      this.props.onSubmit({
        startOffLocation,
        dropOffLocation,
      })
    }
  }

  onFocus = () => {
    this.setState(() => ({ disabled: false }));
  }

  render() {
    const { startOffLocation, error, dropOffLocation, disabled } = this.state;

    return (
      <form className="form" onSubmit={this.onSubmit}>
        {this.state.error && <p className="form__error">{this.state.error}</p>}
        <input
          className="text-input"
          onFocus={this.onFocus}
          placeholder="Enter start off location"
          type="text"
          value={this.state.startOffLocation}
          onChange={this.onStartOffLocationChange}
        />
        <input
          className="text-input"
          onFocus={this.onFocus}
          placeholder="Enter drop off location"
          type="text"
          value={this.state.dropOffLocation}
          onChange={this.onDropOffLocationChange}
        />
        <button
          disabled={!startOffLocation || !dropOffLocation || disabled}
          className="button"
          type="submit"
        >
          Send Location
        </button>
      </form>
    );
  }
}

