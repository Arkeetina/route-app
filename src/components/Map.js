import React, { Component } from 'react';
import {
  withGoogleMap,
  GoogleMap,
  DirectionsRenderer,
} from 'react-google-maps';

class Map extends Component {
  constructor(props) {
    super(props);
    this.state = {
      directions: '',
    };
  }

  componentWillReceiveProps(nextProps) {
    const DirectionsService = new google.maps.DirectionsService();

    DirectionsService.route({
      // не круть, вообще не круть
      origin: new google.maps.LatLng(nextProps.path[0][0], nextProps.path[0][1]),
      waypoints: [{location: `${nextProps.path[1][0]}, ${nextProps.path[1][1]}`,}],
      destination: new google.maps.LatLng(nextProps.path[2][0], nextProps.path[2][1]),
      travelMode: google.maps.TravelMode.DRIVING,
    }, (result, status) => {
      if (status === google.maps.DirectionsStatus.OK) {
        this.setState({
          directions: result,
        });
      } else {
        console.error(`error fetching directions ${result}`);
      }
    });
  }

  render() {
    return (
      <GoogleMap
        defaultZoom={12}
        defaultCenter={new google.maps.LatLng(22.28552, 114.15769)}
      >
        {this.state.directions && <DirectionsRenderer directions={this.state.directions} />}
      </GoogleMap>
    );
  }
}

export default withGoogleMap(Map);
