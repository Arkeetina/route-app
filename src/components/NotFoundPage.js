import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
  <div className="box-container">
    <div className="box-layout">
      <div className="box-layout__box">
        404 - <Link className="link" to="/">Go home</Link>
      </div>
    </div>
  </div>
);

export default NotFoundPage;
