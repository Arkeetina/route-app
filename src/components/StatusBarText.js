import React from 'react';
import { Link } from 'react-router-dom';
import uuid from 'uuid';

const StatusBarText = ({
  defaultMsg,
  message,
  distance,
  time
}) => (
  <div>
    {message && <p className="status-bar__text" key={uuid()}>{message}</p>}
    {time && <p className="status-bar__text" key={uuid()}>Travel time {Math.round(time / 60)} min.</p>}
    {distance && <p className="status-bar__text" key={uuid()}>Total distance {distance / 1000} km</p>}
  </div>
);


export default StatusBarText;
