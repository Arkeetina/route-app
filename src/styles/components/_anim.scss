.anim-appear {
  opacity: 0.01;
}
 
.anim-appear.anim-appear-active{
  opacity: 1;
  transition: opacity 1s ease-in;
}

.anim-leave {
  opacity: 1;
}

.anim-leave.anim-leave-active {
  opacity: 0.01;
  transition: opacity 1s ease-in;
}