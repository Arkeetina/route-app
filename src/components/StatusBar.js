import React, { Component } from 'react';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
import uuid from 'uuid';

import StatusBarText from './StatusBarText';

class StatusBar extends Component {
  loaderRender = () => {
    const { loading } = this.props;
    if (loading) {
      return <img className="loader__image" src="/images/loader.gif"/>;
    }
  }

  defaultMsgRender = () => {
    const { defaultMsg } = this.props;
    if (defaultMsg) {
      return <StatusBarText message={defaultMsg}/>;
    }
  }

  routeInProgStatusRender = () => {
    const { routeInProgStatus } = this.props;
    if (routeInProgStatus) {
      return <StatusBarText message={routeInProgStatus}/>;
    }
  }

  routeFailStatusRender = () => {
    const { routeFailStatus } = this.props;
    if (routeFailStatus) {
      return <StatusBarText message={routeFailStatus}/>;
    }
  }

  serverErrorRender = () => {
    const { serverError } = this.props;
    const message = "Connection Error, please try again later";

    if (serverError) {
      return <StatusBarText message={message}/>;
    }
  }

  pathRender = () => {
    const { totalTime, totalDistance } = this.props;
    if (totalDistance && totalTime) {
      return <StatusBarText distance={totalDistance} time={totalTime}/>;
    }
  }
  
  render() {
    return (
      <div key={uuid()} className="status-bar status-bar__content">
        <ReactCSSTransitionGroup
          transitionName="anim"
          transitionEnterTimeout={500}
          transitionAppear
          transitionAppearTimeout={1000}
          transitionLeaveTimeout={1000}
        >
          {this.defaultMsgRender()}
          {this.routeInProgStatusRender()}
          {this.routeFailStatusRender()}
          {this.serverErrorRender()}
          {this.pathRender()}
          {this.loaderRender()}
        </ReactCSSTransitionGroup>
      </div>
    );
  }
}

export default StatusBar;
