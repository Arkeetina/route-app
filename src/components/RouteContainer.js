import React, { Component } from 'react';
import { connect } from 'react-redux';
import { startSetRoute } from '../actions/location';
import LocationForm from './LocationForm';
import Map from './Map';
import StatusBar from './StatusBar';

const GOOGLE_URL = "https://maps.googleapis.com/maps/api/js?key=AIzaSyDKpzSptr4Y7IikRh3vEaGtrFFB8IfQDg8";

export class RouteContainer extends Component {
  onSubmit = (locations) => {
    this.props.startSetRoute(locations);
  };

  render() {
    const {
      defaultMsg,
      path,
      routeInProgStatus,
      routeFailStatus,
      serverError,
      totalDistance,
      totalTime,
    } = this.props;

    return (
      <div className="box-container">
        <StatusBar
          defaultMsg={defaultMsg}
          routeFailStatus={routeFailStatus}
          routeInProgStatus={routeInProgStatus}
          serverError={serverError}
          totalTime={totalTime}
          totalDistance={totalDistance}
        />
        <div className="box-layout">
          <div className="box-layout__box">
            <LocationForm
              onSubmit={this.onSubmit}
            />
            <Map
              googleMapURL={GOOGLE_URL}
              containerElement={<div style={{ height: `400px` }} />}
              mapElement={<div style={{ height: `100%` }} />}
              path={path}
            />
          </div>
        </div>
      </div>
    );
  }
}
const mapDispatchToProps = (dispatch, props) => ({
  startSetRoute: (startOffLocation, dropOffLocation) => dispatch(startSetRoute(startOffLocation, dropOffLocation))
});

const mapStateToProps = (state) => {
  return {
    defaultMsg: state.location.defaultMsg,
    path: state.location.path,
    routeFailStatus: state.location.routeFailStatus,
    routeInProgStatus: state.location.routeInProgStatus,
    serverError: state.location.serverError,
    totalDistance: state.location.totalDistance,
    totalTime: state.location.totalTime,
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(RouteContainer);
