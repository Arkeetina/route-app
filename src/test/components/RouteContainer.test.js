import React from 'react';
import { shallow } from 'enzyme';
import { RouteContainer } from '../../components/RouteContainer';

test('should render RouteContainer correctly', () => {
  const wrapper = shallow(<RouteContainer />);
  expect(wrapper).toMatchSnapshot();
});
