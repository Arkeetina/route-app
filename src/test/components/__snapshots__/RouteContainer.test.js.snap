// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RouteContainer correctly 1`] = `
<div
  className="box-container"
>
  <StatusBar />
  <div
    className="box-layout"
  >
    <div
      className="box-layout__box"
    >
      <LocationForm
        onSubmit={[Function]}
      />
      <withGoogleMap(Map)
        containerElement={
          <div
            style={
              Object {
                "height": "400px",
              }
            }
          />
        }
        googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyDKpzSptr4Y7IikRh3vEaGtrFFB8IfQDg8"
        mapElement={
          <div
            style={
              Object {
                "height": "100%",
              }
            }
          />
        }
      />
    </div>
  </div>
</div>
`;
